### PR TITLE
fix(agw): Converted ha task files to cpp

### PR DESCRIPTION
**Pull request by VinashakAnkitAman**
## Title
fix(agw): Converted ha task files to cpp  [Extension to Github Issue: #13096]

## Summary
Migrated all the files related to ha task in the folders `magma/lte/gateway/c/core/oai/tasks/ha/` and `magma/lte/gateway/c/core/oai/include/` to `cpp/hpp` and fixed all the compilation errors and warnings.

Also followed the guidelines for order of headers from https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes

## Test plan
Verified compilation with following:
make clean; make build_oai;
make clean; make test_oai;
make clean; make test_oai BUILD_TYPE=Debug OAI_TESTS=test_amf_app
make clean; bazel build --config=production //lte/gateway/c/core:agw_of
make clean; bazel test //lte/gateway/c/core/oai/test/amf:amf_stateless_test
make clean; make FEATURES=mme_oai build_oai
make clean; export FEATURES=mme_oai; make build_oai

Verified with sanity and in CI

Signed-off-by: VinashakAnkitAman <ankit.aman@radisys.com>

### CODE REVIEW

File: lte/gateway/c/core/oai/tasks/mme_app/mme_app_ha.cpp [hunk 1]
```diff
@@ -18,7 +18,7 @@
 extern "C" {
 #include "lte/gateway/c/core/oai/common/log.h"
 #include "lte/gateway/c/core/oai/tasks/mme_app/mme_app_ha.hpp"
-#include "lte/gateway/c/core/oai/include/ha_messages_types.h"
+#include "lte/gateway/c/core/oai/include/ha_messages_types.hpp"
```
Comment (pruthvihebbani) [inline lte/gateway/c/core/oai/tasks/mme_app/mme_app_ha.cpp]: Move this out of extern block

**Review by pruthvihebbani — CHANGES_REQUESTED** [review]

File: lte/gateway/c/core/oai/tasks/mme_app/mme_app_ha.cpp (hunk 1 above)
Comment (VinashakAnkitAman) [reply lte/gateway/c/core/oai/tasks/mme_app/mme_app_ha.cpp]: Moving this newly converted file was fine. Along with this file, I moved `#include "lte/gateway/c/core/oai/tasks/mme_app/mme_app_ha.hpp"` also out of extern, but that was giving compilation error for undefined reference to function `mme_app_handle_ue_offload` which is used in the file `tasks/mme_app/mme_app_main.c`, therefore I have moved the function `mme_app_handle_ue_offload` prototype under extern c in the file `tasks/mme_app/mme_app_ha.hpp`

**Review by LKreutzer — APPROVED** [review]
Bazel changes lgtm

**Review by pruthvihebbani — APPROVED** [review]
LGTM

**Pull request merged**